src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1591,9 +1591,15 @@ static int client_parse_message(
         case DHCP_STATE_REQUESTING:
         case DHCP_STATE_RENEWING:
         case DHCP_STATE_REBINDING:
-                if (r == DHCP_NAK)
+                if (r == DHCP_NAK) {
+                        if ((client->lease != NULL) &&
+                            (client->lease->server_address != lease->server_address))
+                                    return log_dhcp_client_errno(client, SYNTHETIC_ERRNO(ENOMSG),
+                                                                 "NAK: %s from other server",
```

Comment:
Please explicitly mention that the message will be ignored.
```c
"NAK: %s from other server, ignoring."
```

Reply:
Thanks, changed.